### PR TITLE
engine: backport some fixes from 0.19 

### PR DESCRIPTION
**Pull request by SNMetamorph**
Too early to merge now.

### CODE REVIEW

**Comment by mittorn** [comment]
Did you test demo recording/playback?

File: engine/client/client.h [hunk 1]
```diff
@@ -844,6 +844,9 @@ movevars_t *pfnGetMoveVars( void );
 
 _inline cl_entity_t *CL_EDICT_NUM( int n )
```
Comment (mittorn) [inline engine/client/client.h:845]: Maybe it should not be called when clgame.entities is NULL?

Comment (SNMetamorph) [reply engine/client/client.h:845]: Yeah, that's better way

Comment (SNMetamorph) [reply engine/client/client.h:845]: Maybe write assert here instead return NULL? Or not assert, but something for to break debugger here

**Comment by SNMetamorph** [comment]
@mittorn yes I tested demo playing/recoring, it's properly working

**Pull request merged**